source/common/matcher/field_matcher.h

```diff
@@ -167,6 +168,15 @@ class SingleFieldMatcher : public FieldMatcher<DataType>, Logger::Loggable<Logge
   }
 
 private:
+  static DataInputPtr<DataType>& validateInputAndMatcher(DataInputPtr<DataType>& data_input,
+                                                         InputMatcherPtr& input_matcher) {
+    auto supported_input_types = input_matcher->supportedDataInputTypes();
+    if (supported_input_types.find(data_input->dataInputType()) == supported_input_types.end()) {
+      throw EnvoyException(
+          absl::StrCat("Unsupported data input type: ", data_input->dataInputType()));
```

Comment:
Given that this error is user visible, this is makes it more important to use human readable string for type identification. It would also be good to include both the expected and offered types.

Reply:
Sounds reasonable, esp for testing and debugging.